pkg/runctl/run_manager.go

```diff
@@ -42,17 +43,25 @@ type RunManager interface {
 }
 
 type runManager struct {
-	secretProvider   secrets.SecretProvider
-	factory          k8s.ClientFactory
-	namespaceManager k8s.NamespaceManager
+	secretProvider        secrets.SecretProvider
+	factory               k8s.ClientFactory
+	namespaceManager      k8s.NamespaceManager
+	uniqueNameTransformer secrets.SecretTransformerType
+	testing               *runManagerTesting
```

Comment:
Hmm, test code in productive code. 
I did not check the details, but wouldn't it be nicer to e.g. pass the `SecretHelper` to be used to the RunManager, instead of this `testing`?

Reply:
We discussed this already. Yes, we could have a `secretHelper` field in the `runManager` struct.

The testing struct is a versatile pattern to make code testable without introducing artificial interfaces. The benefits here are:

- the object does not exist all the time but only when needed
- the main struct is not polluted by things needed by individual methods

In general:

- avoid artificial interfaces for the sole purpose of testing
- avoid mock generation
- stubs can be easily defined within test methods (function values), while defining a type fulfilling an interface must be done in package scope

Reply:
Should be keep it as is, or have a `secretHelper` field instead?

Reply:
We keep it as is.